CMakeLists.txt

```diff
@@ -827,9 +814,55 @@ if(UNIX)
     find_package(PkgConfig REQUIRED)
 endif()
 
-if (OPENSIM_WITH_CASADI OR OPENSIM_WITH_TROPTER)
+if (OPENSIM_WITH_CASADI)
     if(UNIX)
         pkg_check_modules(IPOPT REQUIRED ipopt IMPORTED_TARGET)
+
+        if(OPENSIM_COPY_DEPENDENCIES AND APPLE)
```

Comment:
I think, of the changes, these additions are the ones that make me go *hmmmmm*.

It looks like you're trying to tell cmake to add those libraries as runtime libraries to the installation set. Fair enough, but `cmake` has functions specifically for doing this, such as `install(TARGETS libOsimMoco RUNTIME_DEPENDENCIES)`.

If cmake's dependency resolution mechanisms don't work, it's usually a sign that your library targets (i.e. the things passed to `target_link_libraries`) aren't declaring which files are dependent on which. Do as you wish, but the way this is written right now smells fishy (e.g. it won't work if the Linux target architecture is changed from `x86_64`, or if it's ran on a distribution of Linux that has a different sys library directory layout).